deepchem/models/torch_models/antibody_modeling.py

```diff
@@ -0,0 +1,273 @@
+            0.7314766049385071) ...
+            ]
+    """
+
+    def __init__(self,
+                 task: str = "mlm",
+                 model_path: str = 'Rostlab/prot_bert',
+                 n_tasks: int = 1,
+                 **kwargs) -> None:
+        """
+        Parameters
+        ----------
+        task: str
+            The task defines the type of learning task in the model. The supported tasks are
+            - `mlm` - masked language modeling commonly used in pretraining
+            - `classification` - use it for classification tasks
+        model_path: str
+            Path to the HuggingFace model
+            - 'Rostlab/prot_bert' - Pretrained on Uniref100 dataset
+            - `Rostlab/prot_bert_bfd` - Pretrained on BFD dataset
+        n_tasks: int
+            Number of prediction targets for a multitask learning model
+        """
+        self.n_tasks: int = n_tasks
+        tokenizer: AutoTokenizer = AutoTokenizer.from_pretrained(
+            model_path, do_lower_casse=False)
+        config: AutoConfig = AutoConfig.from_pretrained(
+            pretrained_model_name_or_path=model_path,
+            vocab_size=tokenizer.vocab_size)
+        model: Union[AutoModel, AutoModelForMaskedLM]
+        if task == "mlm":
+            model = AutoModelForMaskedLM.from_pretrained(model_path)
+        else:
+            model = AutoModel.from_pretrained(model_path)
+        super().__init__(model=model, task=task, tokenizer=tokenizer, **kwargs)
+
+    def _mask_seq_pos(
+        self,
+        sequence: str,
+        idx: int,
+    ):
+        '''Given an arbitrary antibody sequence with and a seqeunce index,
+        convert the residue at that index into the mask token.
+
+        Parameters
+        ----------
+        sequence: str
+            The sequence to be masked at a sepcific residue
+        idx: int
+            The 0-indexed integer index in which into mask the sequence
+
+        Returns
+        -------
+        masked_sequence: str
+            A nearly identical sequence to the input sequence with the token
+            at position idx+1 being the mask token.
+        '''
+        assert isinstance(idx,
+                          int), f"Index must be an int, got type {type(idx)}"
+        assert abs(idx) < len(sequence), f"Zero-indexed idx needs to \
+            be less than {len(sequence)-1} for sequence of length {len(sequence)}."
+
+        cleaned_sequence = sequence.replace(
+            ' ', '')  # Get ride of extraneous spaces if any
+        cleaned_sequence = list(
+            cleaned_sequence)  # Turn the sequence into a list
+        cleaned_sequence[idx] = '*'  # Mask the sequence at idx
+        masked_sequence = ' '.join(cleaned_sequence)  # Convert list -> seq
+        masked_sequence = masked_sequence.replace('*',
+                                                  self.tokenizer.mask_token)
+        return masked_sequence
+
+    def redesign_residue(self,
```

Comment:
We need unit tests

Reply:
Unit tests and integration tests have added